main.go

```diff
@@ -169,26 +175,32 @@ func main() {
 	}
 }
 
-func prepareEventsToTrace(eventsToTrace []string, excludeEvents []string) ([]int32, error) {
+func prepareEventsToTrace(eventsToTrace []string, tagsToTrace []string, excludeEvents []string) ([]int32, error) {
 	var res []int32
 	eventsNameToID := make(map[string]int32, len(tracee.EventsIDToEvent))
 	for _, event := range tracee.EventsIDToEvent {
 		eventsNameToID[event.Name] = event.ID
 	}
-	if eventsToTrace == nil {
+	tagsToEvents := make(map[string][]int32)
+	for id, event := range tracee.EventsIDToEvent {
+		for _, tag := range event.Tags {
+			tagsToEvents[tag] = append(tagsToEvents[tag], id)
+		}
+	}
+	if eventsToTrace == nil && tagsToTrace == nil {
+		isExcluded := make(map[int32]bool)
 		for _, name := range excludeEvents {
 			id, ok := eventsNameToID[name]
 			if !ok {
 				return nil, fmt.Errorf("invalid event to exclude: %s", name)
 			}
-			event := tracee.EventsIDToEvent[id]
-			event.EnabledByDefault = false
-			tracee.EventsIDToEvent[id] = event
+			isExcluded[id] = true
 		}
 		res = make([]int32, 0, len(tracee.EventsIDToEvent))
-		for _, event := range tracee.EventsIDToEvent {
-			if event.EnabledByDefault {
-				res = append(res, event.ID)
+		defaultEvents, _ := tagsToEvents["default"]
+		for _, id := range defaultEvents {
+			if !isExcluded[id] {
```

Comment:
event exclusion is skipped if specifying a tag. need to include this in the lines 215-223 block

Reply:
This is on purpose. It's like "--events" and "--exclude-events" can be used on the same time. If the user chose a tag, it's like choosing events

Reply:
The problem at first place of having both included and excluded events is that we will need to handle the cases where the user may give both an event to include (directly or by tag), and give the same event to exclude. This will require to resolve the conflict, or alert the user. We can do such additional checks and allow this behavior in a different commit

Reply:
I don't think we need conflict resolution or anything similar, exclusion should always win, so if you just [apply the event exclusion here as well](https://github.com/aquasecurity/tracee/pull/198#discussion_r467547780) then it will just filter out unwanted eventswork as expected right?

Reply:
This is one kind of conflict resolution - that exclude always wins. But why is it reasonable that the user will both say "I want to include open event, but also exclude open event"? This is definetly a problem.
I think we should tell the user that there is a conflict and exit - but as I said, we can do it in a different commit

Reply:
I disagree but we can still merge :)

Reply:
I already changed it per your request (before your last comment)